### PR TITLE
Add the ability to add custom labels to log entries.

### PR DESCRIPTION
**Pull request by iantalarico**
This will allow users to add special labels and tags to logs.

### CODE REVIEW

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs [hunk 1]
```diff
@@ -45,16 +52,19 @@ private LoggerOptions(LogLevel logLevel, MonitoredResource monitoredResource, Bu
...
+        /// <param name="labels">Optional, custom labels to be added to log entries.</param>
         /// <param name="monitoredResource">Optional, the monitored resource.  The monitored resource will
         ///     be automatically detected if it is not set and will default to the global resource if the detection fails.
         ///     See: https://cloud.google.com/logging/docs/api/v2/resource-list </param>
         /// <param name="bufferOptions">Optional, the buffer options.  Defaults to a <see cref="BufferType.Timed"/></param>
         public static LoggerOptions Create(LogLevel logLevel = LogLevel.Information,
+            Dictionary<string, string> labels = null,
             MonitoredResource monitoredResource = null, BufferOptions bufferOptions = null)
         {
+            labels = labels ?? new Dictionary<string, string>();
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs]: Are we calling this often? I wouldn't want to create an empty dictionary for no reason if we don't need to.

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs [hunk 2]
```diff
@@ -95,13 +95,14 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
...
             LogEntry entry = new LogEntry
-            {   
+            {
                 Resource = _loggerOptions.MonitoredResource,
                 LogName = _logName,
                 Severity = logLevel.ToLogSeverity(),
                 Timestamp = Timestamp.FromDateTime(_clock.GetCurrentDateTimeUtc()),
                 TextPayload = message,
             };
+            entry.Labels.Add(_loggerOptions.Labels);
```
Comment (jskeet) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs]: Just add

```csharp
Labels = { _loggerOptions.Labels }
```

in the object initializer

Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs]: Done

File: apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs (hunk 1 above)
Comment (iantalarico) [inline apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/LoggerOptions.cs]: This is called once at start up.

**Pull request merged**